.github/workflows/actions/deploy-environment/action.yml

```diff
@@ -0,0 +1,128 @@
+name: Deploy to Azure
+
+inputs:
+  environment_name:
+    description: "The name of the environment"
+    required: true
+  image_tag:
+    description: "github SHA of the image"
+    required: true
+  azure_credentials:
+    description: "JSON object containing a service principal that can read from Azure Key Vault"
+    required: true
+  terraform_vars:
+    description: "Path to the tfvars file for the environment"
+    required: true
+  terraform_backend_vars:
+    description: "Path to the backend tfvars file for the environment"
+    required: true
+
+outputs:
+  environment_url:
+    description: "The base URL for the deployed environment"
+    value: ${{ steps.terraform.outputs.url }}
+  postgres_server_name:
+    description: The name of the postgres server deployed
+    value: ${{ steps.terraform.outputs.postgres_server_name }}
+
+runs:
+  using: composite
+
+  steps:
+    - name: Extract configuration from tfvars
+      id: config
+      run: |
+        KEY_VAULT_NAME=$(jq -r '.key_vault_name' $TFVARS)
+        RESOURCE_GROUP_NAME=$(jq -r '.resource_group_name' $TFVARS)
+        RESOURCE_PREFIX=$(jq -r '.resource_prefix' $TFVARS)
+
+        if [ -z "$KEY_VAULT_NAME" ]; then
+          echo "::error ::Failed to extract key_vault_name from $TFVARS"
+          exit 1
+        fi
+
+        if [ -z "$RESOURCE_GROUP_NAME" ]; then
+          echo "::error ::Failed to extract resource_group_name from $TFVARS"
+          exit 1
+        fi
+
+        if [ -z "$RESOURCE_PREFIX" ]; then
+          echo "::error ::Failed to extract resource_prefix from $TFVARS"
+          exit 1
+        fi
+
+        echo ::set-output name=resource_group_name::$RESOURCE_GROUP_NAME
+        echo ::set-output name=key_vault_name::$KEY_VAULT_NAME
+        echo ::set-output name=resource_prefix::$RESOURCE_PREFIX
+
+      shell: bash
+      env:
+        TFVARS: ${{ inputs.terraform_vars }}
+      working-directory: terraform
+
+    - uses: Azure/login@v1
+      with:
+        creds: ${{ inputs.azure_credentials }}
+
+    - uses: Azure/get-keyvault-secrets@v1
```

Comment:
Is it worth changing this to use the `[azure/cli@v1](https://github.com/Azure/cli)` action as the [get-keyvault-secrets](https://github.com/Azure/get-keyvault-secrets) action is deprecated now?